server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/measure/BestValueOptimization.java

```diff
@@ -60,20 +60,24 @@ private boolean isBestValueOptimized(Measure measure) {
     return measure.getDescription() == null
       && measure.getData() == null
       && !measure.hasQualityGateStatus()
-      && hasNoVariation(measure)
+      && hasNoVariation(measure, metric.getBestValue())
```

Comment:
so we never check the best value!?! surprising, but as you said it has always been 0.
for coverage, we don't use it? because the best value would be 100%